sktime/base/_base.py

```diff
@@ -323,6 +323,40 @@ def create_test_instances_and_names(cls):
 
         return objs, names
 
+    @classmethod
+    def _has_implementation_of(cls, method):
+        """Check if method has a concrete implementation in this class.
+
+        This assumes that having an implementation is equivalent to
+            one or more overrides of `method` in the method resolution order.
+
+        Parameters
+        ----------
+        method : str, name of method to check implementation of
+
+        Returns
+        -------
+        bool, whether method has implementation in cls
+            True if cls.method has been overridden at least once in
+                the inheritance tree (according to method resolution order)
+        """
+        # walk through method resolution order and inspect methods
+        #   of classes and direct parents, "adjacent" classes in mro
+        mro = inspect.getmro(cls)
+        # collect all methods that are not none
+        methods = [getattr(c, method, None) for c in mro]
+        methods = [c for c in methods if c is not None]
+
+        for i in range(len(methods) - 1):
```

Comment:
how about using a `return len(set(methods)) >= 2 `  instead of this block?

Reply:
I'm not sure whether that's "safe" since we're operating with pointers here?